pkg/controller/spinnakerservice/status.go

```diff
@@ -3,62 +3,69 @@ package spinnakerservice
 import (
 	"context"
 
+	spinnakerv1alpha1 "github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha2"
 	appsv1 "k8s.io/api/apps/v1beta2"
-
-	// corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/selection"
-
-	spinnakerv1alpha1 "github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type statusChecker struct {
 	client client.Client
 }
 
+const (
+	Ok          = "OK"
+	Updating    = "Updating"
+	Unavailable = "Unavailable"
+	Na          = "N/A"
+)
+
 func newStatusChecker(client client.Client) statusChecker {
 	return statusChecker{client: client}
 }
 
-type labelSelector struct{}
-
-func (l *labelSelector) Matches(labels labels.Labels) bool {
-	return labels.Get("app.kubernetes.io/managed-by") == "halyard"
-}
-
 func (s *statusChecker) checks(instance spinnakerv1alpha1.SpinnakerServiceInterface) error {
-	r, err := labels.NewRequirement("app.kubernetes.io/managed-by", selection.Equals, []string{"halyard"})
-	if err != nil {
-		return err
-	}
-	sel := labels.NewSelector()
-	sel.Add(*r)
 	// Get current deployment owned by the service
 	list := &appsv1.DeploymentList{}
-	err = s.client.List(context.TODO(), &client.ListOptions{LabelSelector: sel, Namespace: instance.GetNamespace()}, list)
+	err := s.client.List(context.TODO(), list, client.InNamespace(instance.GetNamespace()), client.MatchingLabels{"app.kubernetes.io/managed-by": "halyard"})
```

Comment:
Maybe this `app.kubernetes.io/managed-by` annotation would be better off having managed-by `spinnaker-operator`?